playwright-junit/src/main/java/com/microsoft/playwright/junit/BrowserExtension.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.playwright.junit;
+
+import com.microsoft.playwright.Browser;
+import com.microsoft.playwright.Playwright;
+import com.microsoft.playwright.PlaywrightException;
+import org.junit.jupiter.api.extension.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.microsoft.playwright.junit.PlaywrightExtension.getOrCreatePlaywright;
+import static org.junit.platform.commons.support.AnnotationSupport.findAnnotation;
+import static org.junit.platform.commons.support.AnnotationSupport.isAnnotated;
+
+public class BrowserExtension implements ParameterResolver, AfterAllCallback {
+  private static final ThreadLocal<Map<Class<? extends BrowserFactory>, Browser>> threadLocalBrowserMap;
```

Comment:
Can there be more than one entry in the map? My understanding is that there is at most one active test class and each class has only one UseBrowserFactory annotation.

Reply:
Good catch. This is a left over from when we would allow more than one browser per class. I'll fix it.